numeric.rb

```diff
@@ -217,6 +217,29 @@ def size
     Primitive.cexpr! 'rb_int_size(self)'
   end
 
+  # call-seq:
+  #   times {|i| ... } -> self
+  #   times            -> enumerator
+  #
+  # Calls the given block +self+ times with each integer in <tt>(0..self-1)</tt>:
+  #
+  #   a = []
+  #   5.times {|i| a.push(i) } # => 5
+  #   a                        # => [0, 1, 2, 3, 4]
+  #
+  # With no block given, returns an Enumerator.
+  def times
+    unless block_given?
+      return to_enum(:times) { self < 0 ? 0 : self }
+    end
+    i = 0
+    while i < self
+      yield i
+      i = i.succ
```

Comment:
This is faster than `i += 1`?

Reply:
Yes, it is for the interpreter. It's `putobject 1` + `opt_plus` vs `opt_succ`, so one less instruction for the interpreter.

Reply:
```
$ benchmark-driver benchmark/loop_times.rb --chruby 'opt_succ::before;opt_plus::after' -v
opt_succ: ruby 3.3.0dev (2023-09-06T23:26:33Z builtin-times a2bee28ecc) [x86_64-linux]
opt_plus: ruby 3.3.0dev (2023-09-07T00:26:06Z builtin-times-plus 24394f9316) [x86_64-linux]
Calculating -------------------------------------
                       opt_succ    opt_plus
          loop_times      1.070       0.955 i/s -       1.000 times in 0.934151s 1.047303s

Comparison:
                       loop_times
            opt_succ:         1.1 i/s
            opt_plus:         1.0 i/s - 1.12x  slower
```